feat(cli): support CloudFormation simplified resource import

tmokmss opened the pull request:
### Issue # (if applicable)

Closes #28060.

### Reason for this change

This feature allows to automatically import exsting resources with the same physical name, such as S3 bucket, DDB table, etc, during a CFn deployment.

Because resource import is a vital feature for CDK users e.g. to refactor a construct tree, cdk migrate, etc, it would benefit many potential users if cdk natively support it.

### Description of changes

This PR adds a CLI option --import-exsting-resources: boolean to cdk deploy command and pass it to createChangeSet API call.

### Description of how you validated changes

Added a cli integ test.

### Checklist
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)



----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation reviewed (changes requested):
The pull request linter has failed. See the aws-cdk-automation comment below for failure reasons. If you believe this pull request should receive an exemption, please comment and provide a justification.


A comment requesting an exemption should contain the text `Exemption Request`. Additionally, if clarification is needed add `Clarification Request` to a comment.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.
	❌ The first word of the pull request title should not be capitalized. If the title starts with a CDK construct, it should be in backticks "``".

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.
	❌ The first word of the pull request title should not be capitalized. If the title starts with a CDK construct, it should be in backticks "``".

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

tmokmss commented:
Exemption Request: I added a CLI integ test instead of framework-integ, because this is a change of CLI.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 8803a585b51c9aac51c391878c39e8521a832778
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=afd3db45-1632-4ad0-9cf1-a5e483dece62%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 1051e8fa3f86e030d9e04486d692f7cb36174e32
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b4b87879-d683-4999-a055-db5f61b456a6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented:
I've been thinking about this one a LOT and been back and forth about what I think the right solution for it is. One issue I have with just adding this flag to `deploy` is the amount of flags we now have on this command that only work with some/none of the other flags. I'm worried that it's getting too messy.

I've also been thinking that with the interplay between `cdk migrate`, `cdk import`, `cfn-include`, and this feature, there's a lot of weird messy overlap that I'd like to potentially clean up a bit. Maaaaaaybe this should actually be added as a flag on `cdk import` instead? 

Right now `cdk import` has some kind of weird limitations that were due to unresolved issues on CloudFormation's end that I got resolved for `cdk migrate` so maybe the right path forward is fixing those oddities and then adding this flag there.

tmokmss commented:
@TheRealAmazonKendra Thanks. I agree with that both `cdk migrate` and `cdk import` commands can internally use the CloudFormation simplified import feature for better UX. But I also think we should definitely add this flag to `cdk deploy` command as the lowest level API to access the feature. 

The fact is that users currently cannot use the feature at all, even though the feature has been available since last year, because there is no way to add argument from outside to `createChangeSet` API called by `deploy` command.

The feature itself is quite simple, and in most use cases, users should not need fancy API around it.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.
	❌ CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin pull/29087/head && git push -f origin FETCH_HEAD:test-main-pipeline), then add the 'pr-linter/cli-integ-tested' label when the pipeline succeeds.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: af43d963412b1cfa88b6e9fbea6bf3c42c8ea529
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=33aee770-b199-4d53-b027-6ccf917b3b38%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
:arrow_right: **PR build request submitted to `test-main-pipeline`** :arrow_left:

A maintainer must now check the pipeline and add the `pr-linter/cli-integ-tested` label once the pipeline succeeds.
<!-- thollander/actions-comment-pull-request "request-cli-integ-test" -->

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

✅ A exemption request has been requested. Please wait for a maintainer's review.

tmokmss commented:
@SankyRed Thanks! I guess we also need to add `pr-linter/exempt-integ-test` label?

comcalvi commented on packages/aws-cdk/lib/api/deploy-stack.ts:
```diff
@@ -416,6 +424,7 @@ class FullCloudFormationDeployment {
       ResourcesToImport: this.options.resourcesToImport,
       Description: `CDK Changeset for execution ${this.uuid}`,
       ClientToken: `create${this.uuid}`,
+      ImportExistingResources: importExistingResources,
```
what happens if they specify both `ResourcesToImport` and `ImportExistingResources`?

tmokmss replied:
I guess it will not work, but [the spec](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_CreateChangeSet.html) says nothing about such restriction, so I would like to keep it as-is. 

Actually users cannot set both ImportExistingResources and ResourcesToImport from the CDK CLI, because the former is only used in cdk deploy, and cdk deploy does not have any command argument to directly set ResourcesToImport.

HBobertz replied:
The spec isn't really the source of truth we all wish it was so it's not uncommon we enforce logical things the spec does not. However in this specific instance I believe this is correct because there is a difference between the resources supported by `importExistingResources` and `ResourcesToImport`. Auto-import is a subset of resources supported by `ResourcesToImport` so it is entirely possible to have both. I believe this is fine to leave as is

comcalvi reviewed (dismissed):
we probably want this feature on diff changesets as well... @TheRealAmazonKendra what do you think?

rachit-shah commented:
@comcalvi @tmokmss @TheRealAmazonKendra 
We use CDK deployment and face resource conflicts often. We'd like to avoid calling CFN APIs directly since we are using CDK app. 

What exactly is this PR blocked on? It has been months since the PR was raised. If it's adding this feature in a different cdk operation (cdk diff), can that be a separate PR/issue?

tmokmss commented:
@comcalvi Hi, I resolved the merge conflict. Can you take a look at this once again?

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c8be1f5e41eb50670b4f296198b3ee9db4dbb49c
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0601d220-6061-442e-812b-93cdb58216cc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

HBobertz commented on packages/aws-cdk/README.md:
````diff
@@ -388,6 +388,40 @@ $ cdk deploy --method=prepare-change-set --change-set-name MyChangeSetName
+$ cdk deploy --import-existing-resources
+```
+
+This allows you to automatically import the resources in your template that already
+exist in your AWS account during CloudFormation deployments.
+With this feature, you can reduce the manual effort of import operations and avoid
+deployment failures because of naming conflicts.
````
```suggestion
Automatically import resources in your CDK application which represent
unmanaged resources in your account. 
Reduces the manual effort of import operations and avoids
deployment failures due to naming conflicts with unmanaged resources in your account
```

HBobertz replied:
adding some active voice to be more similar to how our other command readme's are written.

HBobertz commented on packages/aws-cdk/README.md:
```diff
@@ -388,6 +388,40 @@ $ cdk deploy --method=prepare-change-set --change-set-name MyChangeSetName
+With this feature, you can reduce the manual effort of import operations and avoid
+deployment failures because of naming conflicts.
+
+To review which resources are imported or not before actually executing a change set,
+use `--method=prepare-change-set` flag.
```
```suggestion
use `--method=prepare-change-set` flag to review which resources are imported or not before deploying
```

HBobertz commented on packages/aws-cdk/README.md:
````diff
@@ -388,6 +388,40 @@ $ cdk deploy --method=prepare-change-set --change-set-name MyChangeSetName
+$ cdk deploy --import-existing-resources --method=prepare-change-set
+```
+
+To enable this feature only for a specific stack, use `--exclusively` flag.
````
```suggestion
Use the `--exclusively` flag to enable this feature for a specific stack
```

HBobertz commented on packages/aws-cdk/README.md:
````diff
@@ -388,6 +388,40 @@ $ cdk deploy --method=prepare-change-set --change-set-name MyChangeSetName
+$ cdk deploy --import-existing-resources --exclusively StackName
+```
+
+This parameter can only import resources that have custom names in templates. 
````
```suggestion
Only resources that have custom names can be imported using `--import-existing-resources`
```

HBobertz commented on packages/aws-cdk/README.md:
```diff
@@ -579,6 +613,11 @@ To import an existing resource to a CDK stack, follow the following steps:
 5. When `cdk import` reports success, the resource is managed by CDK. Any subsequent
    changes in the construct configuration will be reflected on the resource.
 
+NOTE: You can also import existing resources by `--import-existing-resources` option of `cdk deploy` command.
```
```suggestion
NOTE: You can also import existing resources by using `--import-existing-resources` flag of the `cdk deploy` command.
```

HBobertz commented on packages/aws-cdk/README.md:
```diff
@@ -579,6 +613,11 @@ To import an existing resource to a CDK stack, follow the following steps:
    changes in the construct configuration will be reflected on the resource.
 
+NOTE: You can also import existing resources by `--import-existing-resources` option of `cdk deploy` command.
+This parameter only works for resources that you can set custom physical names,
```
```suggestion
Only works for resources that allow custom physical names
```

HBobertz commented on packages/aws-cdk/README.md:
```diff
@@ -579,6 +613,11 @@ To import an existing resource to a CDK stack, follow the following steps:
 
+NOTE: You can also import existing resources by `--import-existing-resources` option of `cdk deploy` command.
+This parameter only works for resources that you can set custom physical names,
+such as S3 bucket, DynamoDB table, etc.
```
```suggestion
such as S3 bucket, DynamoDB table, etc...
```

HBobertz replied:
Love that you're pushing this change and really appreciate the contribution. I've wanted this functionality for a while so it's great to see it being implemented. Just have some doc changes and it's g2g imo

comcalvi commented on packages/aws-cdk/test/api/deploy-stack.test.ts:
```diff
@@ -880,6 +880,41 @@ describe('disable rollback', () => {
 
 });
 
+describe('import-existing-resources', () => {
```
Can we get a CLI integ test that creates a single resource and then imports it with this new flag?

tmokmss replied:
@comcalvi 
Thanks, but I don't fully agree with implementing such test on CDK side. Because:

1. Such test would more or less complicate the test code, not noly the success case, but also ensuring there is no orphaned resource when test failed halfway through
2. It is not CDK's responsibility to ensure that a resource is successfully imported when the flag is set. It is CFn's responsibility to guarantee the functionality. CDK should just make sure the flag is set properly (as tested in the current code.)

What do you think?

HBobertz replied:
> It is not CDK's responsibility to ensure that a resource is successfully imported when the flag is set. It is CFn's responsibility to guarantee the functionality. CDK should just make sure the flag is set properly (as tested in the current code.)

This is true, however ultimately any test involving any cloudformation api usage (such as most CDK deploy tests) also falls under that same umbrella. Even though we don't own cloudformation nor are we responsible to fix any issue blocking our service in cloudformation, it does allow us to report the impact our customer's impact to appropriate CFN teams. Regardless of all that though it's already a just line in the sand we've crossed ages ago, so I believe it's appropriate to add that test

> Such test would more or less complicate the test code, not noly the success case, but also ensuring there is no orphaned resource when test failed halfway through. This possibly increases maintenance cost of the cli integ test code.

This is true, it would be far from the most complex integ test we've written but any integ tests which are verifying deployments tend to require some setup/cleanup so it does get a bit complex. Orphaning is technically possible but as long as you set up a `finally` to manually cleanup you manually created, and the stack rolls back, it should be g2g. You can look at other tests in CDK Deploy or perhaps CDK Migrate for some ideas of how we've done this in the past.

HBobertz replied:
This is ultimately just a very long winded way of me saying I believe the test is appropriate 😅.

tmokmss replied:
@HBobertz Thanks, hmmm but what is the actual value to adding the test? As long as we set the importExistingResources flag, it should result in a successfull deployment, unless CFn is not working properly. If there happens to be the case when CDK fails although CFn is working as expected, that is when we should add a new test for the very case. At this moment, however, I think such test does not add any reliability to the behavior of CDK; it just adds unneessary complexity to the cli integ test code base.

tmokmss commented:
@HBobertz @comcalvi 

Thanks! I addressed all the comments from you :)

> It's unclear if this flag works with IMPORT type changesets. Those changesets should not create anything, but autoimport likely needs a CREATE to import the resource...maybe IMPORT changesets recognize that?

As far as I tested, CFn decides whether a resource is created or imported when we create a changeset. If a resource will be imported, the changeset type for the resource will become `IMPORT`, and if not, it wil become `CREATE`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 8eb154961ec20a10d896dc7e4e5350242f476215
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b1571176-a27e-486e-ad4a-735e9aedffaa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 293a61f951202b639ceacd6537abb1a81243a629
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a6d4db6d-82a0-480e-a28b-5a7549fb7cca%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ced4fb11eff02b1190bd9aae617adbad39bf5e05
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=25d8d6fe-2f5c-425e-93e4-b4bb3cdd217c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

comcalvi reviewed (commented):
LGTM, pending a CLI test run.

tmokmss commented:
@comcalvi Thanks! Just for confirmation, have you triggered a cli test? I see the request is still pending. (not really sure how it works though)
![image](https://github.com/user-attachments/assets/9fd78d03-e4b8-4b27-a0ae-82a06afde547)

aws-cdk-automation commented:
:arrow_right: **PR build request submitted to `test-main-pipeline`** :arrow_left:

A maintainer must now check the pipeline and add the `pr-linter/cli-integ-tested` label once the pipeline succeeds.
<!-- thollander/actions-comment-pull-request "request-cli-integ-test" -->

aws-cdk-automation commented:
:arrow_right: **PR build request submitted to `test-main-pipeline`** :arrow_left:

A maintainer must now check the pipeline and add the `pr-linter/cli-integ-tested` label once the pipeline succeeds.
<!-- thollander/actions-comment-pull-request "request-cli-integ-test" -->

tmokmss commented:
@comcalvi Hi, let me know the integ-test result when you get a chance :) I guess I don't have access to it.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5b1db10f52ba8a4afb97cc90c986f528a2d8d1e9
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=859a2cf8-798e-47e9-a3e5-c77f157f931d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
:arrow_right: **PR build request submitted to `test-main-pipeline`** :arrow_left:

A maintainer must now check the pipeline and add the `pr-linter/cli-integ-tested` label once the pipeline succeeds.
<!-- thollander/actions-comment-pull-request "request-cli-integ-test" -->

kaizencc commented:
@tmokmss I mean to help you get this PR across the finish line. first step was to fix the merge conflicts, which are a little bit difficult. I did it locally, but I'm not able to push to this branch. Is there any reason that is the case on your end? It makes me sad because I spent some time trying to fix up the branch for you :(.

i can also start a new branch and we can go from there but im first wondering if there's something quick you can do on your end to ensure that i have push access to your fork.

tmokmss commented:
Hi @kaizencc, thanks for working on this and sorry I didn't realize it was in conflict.

I confirmed that `Allow edits and access to secrets by maintainers' is checked, so I think you should be able to push commits. Not sure about the error, but if this is a blocker, it's okay to create a new branch and PR. I'm excited to see this finally merged :)

kaizencc commented:
@tmokmss i can't figure it out :/ so i just went ahead and created a new PR to main from your PR. I think you'll still get credit, but i also want to add a coauthor message to the PR description just in case. I'll need an email address associated to your github account to make that happen though.

tmokmss commented:
@kaizencc thanks for your kindness! You can use this address:
tmokmss@users.noreply.github.com

kaizencc commented:
https://github.com/aws/aws-cdk/pull/32676 was merged
